fortex/health/processors/scispacy_processor.py

```diff
@@ -0,0 +1,170 @@
+        if self.configs.pipe_name == "abbreviation_detector":
+            self.extractor.add_pipe(self.configs.pipe_name)
+        else:  # hyponym
+            self.extractor.add_pipe(
+                self.configs.pipe_name, last=True, config={"extended": False}
+            )
+
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        self.set_up()
+
+    def _process(self, input_pack: DataPack):
+        r"""
+        ScispaCyProcessor is done on the basis of
+        using SciSpacy and the corresponding
+        trained model for Hyponym, Abbreviation
+
+        """
+
+        path_str, module_str = self.configs.entry_type.rsplit(".", 1)
+
+        mod = importlib.import_module(path_str)
+        entry = getattr(mod, module_str)
+        for entry_specified in input_pack.get(entry_type=entry):
+
+            doc = self.extractor(entry_specified.text)
+
+            if self.configs.pipe_name == "abbreviation_detector":
+                list_of_abrvs = []
+                for abrv in doc._.abbreviations:
+                    print(
+                        f"{abrv} \t ({abrv.start}, {abrv.end}) {abrv._.long_form}"
+                    )
+                    tmpAbrv = Abbreviation(
+                        pack=input_pack, begin=abrv.start, end=abrv.end
+                    )
+                    tmpAbrv.long_form = abrv._.long_form
+                    list_of_abrvs.append(tmpAbrv)
+
+            else:
+                for item in doc._.hearst_patterns:
+                    general_concept: Phrase = Phrase(
+                        pack=input_pack, begin=item[1].start, end=item[1].end
+                    )
+                    specific_concept = Phrase(
+                        pack=input_pack, begin=item[2].start, end=item[2].end
+                    )
+                    hlink = Hyponym(
+                        pack=input_pack,
+                        parent=general_concept,
+                        child=specific_concept,
+                    )
+                    hlink.hyponym_link = item[0]
+
+    @classmethod
+    def default_configs(cls):
+        r"""
+        This defines a basic config structure for `ScispaCyProcessor`.
+
+        Following are the keys for this dictionary:
+         - `entry_type`: input entry type for classification
+         - `model_name`: the ScispaCy model name to be
+                         used for classification,
+         - `pipe_name`: the Spacy model pipe name to be
+                         used for classification,
+
+        Returns: A dictionary with the default config for this processor.
+        """
+        return {
+            "entry_type": "ft.onto.base_ontology.Document",
+            "attribute_name": "classification",
+            "multi_class": True,
+            "model_name": "en_core_sci_sm",
+            "pipe_name": "abbreviation_detector",
+            "cuda_devices": -1,
```

Comment:
This is also not used in the system, maybe you can follow [here](https://github.com/asyml/forte-wrappers/blob/main/src/spacy/fortex/spacy/spacy_processors.py#L235) for GPU setups.

Reply:
removed and tested ok.